### PR TITLE
Added optional timeout parameter to the job decorator.

### PR DESCRIPTION
**Pull request by samzscott**
Added support for an optional timeout parameter to the job decorator so one may override the redis default of 180 seconds. Does not affect past implementations that use flask-rq. Added an example and entry to the readme accordingly.

### CODE REVIEW

File: flask_rq.py [hunk 1]
```diff
@@ -84,11 +84,15 @@ def job(func_or_queue=None):
     def wrapper(fn):
         def delay(*args, **kwargs):
             q = get_queue(queue)
-            return q.enqueue(fn, *args, **kwargs)
+            if timeout is None:
+                return q.enqueue(fn, *args, **kwargs)
+            else:
+                return q.enqueue_call(func=fn, args=args, kwargs=kwargs, timeout=timeout)
```
Comment (mschmo) [inline flask_rq.py]: Should be using the `default_timeout` argument name since RQ 0.3.0 - http://python-rq.org/docs/results/

**Comment by rkubik** [comment]
Bump. This is very useful, any progress on this?

@mschmo I don't think so. `default_timeout` is for the whole queue not individual jobs.

**Comment by mschmo** [comment]
@rkubik I reread the documentation in the link I posted and I agree with you. Not sure what I was thinking at the time, but that was 9 months ago so who knows.

File: flask_rq.py (hunk 1 above)
Comment (joostdevries) [inline flask_rq.py]: `enqueue` also has a timeout param so I don't think this if statement is needed.

Comment (vaidsu) [inline flask_rq.py]: I need this change too. I was adding it myself in my code base. 

I see the PR is dead due to a connection failure https://travis-ci.org/mattupstate/flask-rq/jobs/29826186#L88

This is very useful. Seems its been a while, may we please resurrect the thread? 

Please let me know if this needs help.

**Comment by cburmeister** [comment]
Would love to see this land on master -- anything I can do to help?

**Pull request closed without merging**